### PR TITLE
Fix entry points data generation

### PR DESCRIPTION
**Pull request by AhmedBasem20**
The issue is that `get_entry_point_names()` function return additional entry points names other than the current plugin entrypoints.
Example:
image
This should only be:
```
{
    "aiida.calculations": [
        "abinit"
    ],
    "aiida.workflows": [
        "abinit.base"
    ]
}
```
My solution is to pass the plugin entry points to the `analyze_entrypoints` module as a click option, and then filter out the calculations and workflow entrypoint instead of using `get_entry_point_names()`.

### CODE REVIEW

File: aiida_registry/test_install.py [hunk 1]
```diff
@@ -182,3 +183,42 @@ def test_install_all(container_image):
...
+        if "pip_url" not in list(plugin.keys()):
+            if plugin["development_status"] not in ["planning", "pre-alpha", "alpha"]:
+                print(
+                    f"    >> WARNING: pip_url key missing, despite required for {plugin['development_status']} stage !"
+                )
+            else:
+                print("    >> SKIPPING: No pip_url key provided")
+            continue
+
+        results = test_install_one_docker(container_image, plugin)
```
Comment (ltalirz) [inline aiida_registry/test_install.py:212]: is this going to try to install all of the plugins again?

**Review by ltalirz — COMMENTED** [review]
thanks for the PR @AhmedBasem20 !

I see your solution but it seems a bit more complex to me than needed.

I first wanted to understand why the problem you noticed exists and, to be honest, I still don't quite understand how it arises.

Looking at the original code, I would expect each plugin install to occur in a separate docker container, so I would expect the return value of one run of `test_install_one_docker` to be entirely independent of the previous one.

I suspect it somehow happens through files being written to the shared working directory - can you please check this by running the workflow locally?

If that is the case, then the "fix" is much easier - simply make sure to clean up these temporary files before running the next container.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -78,6 +78,8 @@ jobs:
 
     - name: Try installing packages
       run: aiida-registry test-install
+    - name: Generate entrypoints data
```
Comment (ltalirz) [inline .github/workflows/ci.yml:81]: I suggest you simply modify the previous step instead.
I see below that you are installing all the plugins again here (which is very slow) - we should only do this once

**Review by ltalirz — CHANGES_REQUESTED** [review]
Thanks @ahmedbasem20 !

The direction is good, but I think there may be an easier way to achieve this

File: aiida_registry/test_install.py [hunk 3]
```diff
@@ -149,12 +155,44 @@ def test_install_one_docker(container_image, plugin):
     )
 
 
+def filter_entry_points(process_metadata, entrypoints):
+    """
+    Extract entry points that belongs to the plugin,
```
Comment (ltalirz) [inline aiida_registry/test_install.py:160]:
```suggestion
    Extract entry points that belong to the plugin,
```

File: aiida_registry/test_install.py [hunk 4]
```diff
@@ -149,12 +155,44 @@ def test_install_one_docker(container_image, plugin):
...
+    and delete any other entry points.
+    """
+
```
Comment (ltalirz) [inline aiida_registry/test_install.py:163]: If I understand correctly, here you are first building up the list of eps, then a list of keys to delete, then delete the keys

how about just _taking_ the entry points you want?

the code below is untested and likely not correct (I don't fully understand the structure of the current dictionaries), but it points in the direction
```
process_metadata = {}
for group_label, group_val in ENTRY_POINT_GROUPS:
  process_metadata[group_label]={}
  for ep_label, _ in process_metadata[group_label].items():
    process_metadata[group_label][ep_label] = group_val[ep_label]
```

File: aiida_registry/test_install.py [hunk 5]
```diff
@@ -177,8 +215,19 @@ def test_install_all(container_image):
...
-    print(f"Dumping {PLUGINS_TEST_RESULTS}")
-    with open(PLUGINS_TEST_RESULTS, "w", encoding="utf8") as handle:
-        json.dump(test_results, handle, indent=2)
+        results = test_install_one_docker(container_image, plugin)
+        process_metadata = results["process_metadata"]
+        for ep_group in ENTRY_POINT_GROUPS:
+            try:
+                if process_metadata[ep_group]:
+                    for key, _ in data[_k]["entry_points"][ep_group].items():
+                        data[_k]["entry_points"][ep_group][key] = process_metadata[
```
Comment (ltalirz) [inline aiida_registry/test_install.py:224]: looks to me like `_k` is being used and should no longer start with an underscore (perhaps `plugin_name` or something)

**Review by ltalirz — COMMENTED** [review]
thank you @AhmedBasem20 , 

the changes you implemented look good to me.

only a few nitpicks left

**Comment by unkcpz** [comment]
replaced by https://github.com/aiidateam/aiida-registry/pull/245

**Pull request closed without merging**